packages/moda-sdk/src/profile/createProfile.ts

```diff
@@ -0,0 +1,33 @@
+import { Hash } from 'viem';
+import { ProfileABI } from './abi/ProfileABI';
+import { profileAddress } from './constants/profileAddress';
+import { CreateProfileArgs } from './types';
+
+export async function createProfile(
+  createProfileArgs: CreateProfileArgs,
+): Promise<Hash> {
+  const { config, metadataUri } = createProfileArgs;
+  const { publicClient, walletClient } = config;
+
+  if (publicClient === undefined) {
+    throw new Error('Public client is undefined');
```

Comment:
It is generally better to return an error vs throwing one. This allows the caller the opportunity to handle it as they want without having to `try / catch`.

Here is one way to do it:

1. Define a type in `utils/types.ts`

```ts
export type Result<T, E = Error> =
  | { ok: true; value: T }
  | { ok: false; error: E };
```

Notice the `E = Error`, which allows us to omit it later and use the default of `Error`. We can optionally change this later.

2. Change the return type in this function:

```ts
export async function createProfile(
  createProfileArgs: CreateProfileArgs,
): Promise<Result<Hash>> {
   // ...
}
```

Now to can do:

```ts
const result = await createProfile(...);

if (!result.ok) {
  console.error(result.error);
  return;
}

console.log(result.value);
```

Reply:
Now let's create better errors with an enum so users can match against known strings without risking misspelling the message.

1. In this file:

```ts
export enum CreateProfileError {
  ClientUndefined = 'Public client is undefined',
  SomeOtherError = 'Foo bar baz',
}
```

2. Change the return type on `createProfile`

```ts
export async function createProfile(
  createProfileArgs: CreateProfileArgs,
): Promise<Result<Hash, CreateProfileError>> {
   // ...
}
```

We override the default `Error` type defined in `Result<T,E>`

Now you can do:

```ts
const result = await createProfile(...);

if (!result.ok) {
  switch result.error {
    case CreateProfileError.ClientError:
      // do something special
      return;
      
    case CreateProfileError.SomeOtherError:
       // do something special
       return;
       
     default:
       // do something else
       return;
  }
}

console.log('everything is fine', result.value);
```